lib/reversetunnel/local_cluster.go

```diff
@@ -634,11 +634,17 @@ func (s *localCluster) getConn(params reversetunnelclient.DialParams) (conn net.
 		return newMetricConn(conn, dialTypeDirect, dialStart, s.srv.Clock), false, nil
 	}
 
+	targetScope := params.TargetScope
+	if params.TargetServer != nil {
+		targetScope = params.TargetServer.GetScope()
+	}
```

Comment:
nit: we may want to verify that scopes are equivalent when they are both defined.  Ex:

```golang
if params.TargetScope != "" && params.TargetServer != nil {
    if scopes.Compare(params.TargetScope, params.TargetServer.GetScope()) != scopes.Equivalent {
        return trace.BadParameter(...)
    }
}
```

idk how this would happen, but ignoring disagreement could mask a security issue, so might be best to treat with caution.